copy/copy.go

```diff
@@ -1166,6 +1166,20 @@ func computeDiffID(stream io.Reader, decompressor compression.DecompressorFunc)
 	return digest.Canonical.FromReader(stream)
 }
 
+// errorAnnotationReader wraps the io.Reader passed to PutBlob for annotating the error happened during read.
```

Comment:
```suggestion
// errorAnnotationReader wraps the io.Reader passed to PutBlob for annotating the error happened during read.
// These errors are reported as PutBlob errors, so we would otherwise misleadingly attribute them to the copy destination.
```
The _why_ is the more difficult part to reverse-engineer from code.

Reply:
If you want this to be merged as-is, that’s fine as well.